app/assets/javascripts/discourse/app/controllers/edit-slow-mode.js

```diff
@@ -66,9 +66,10 @@ export default Controller.extend(ModalFunctionality, {
     }
   },
 
-  @discourseComputed("hours", "minutes", "seconds")
-  submitDisabled(hours, minutes, seconds) {
-    return this.saveDisabled || !(hours || minutes || seconds);
+  @discourseComputed("hours", "minutes", "seconds", "enabledUntil")
+  submitDisabled(hours, minutes, seconds, enabledUntil) {
+    const durationIsSet = hours || minutes || seconds;
```

Comment:
maybe you could have a computed property durationIsSet this way we avoid having so many arguments here, more than 3 is kinda a code smell

Reply:
I've fixed this.